docs/API.md

```diff
@@ -998,7 +1000,7 @@ An object with the following properties:
     `"ERROR"`, but none with `"CRITICAL"`.
   * `"critical"`, if there is at least one message with *severity level*
     `"CRITICAL"`.
-
+* `"undelegated"`: `1` if the test is undelegated, `0` otherwise.
```

Comment:
This is a breaking change. My suggestion was to add a new value on "filter" that would then include `undelegated` in the response. That would not be breaking.

Would that be expensive? What are the downsides of that?

Reply:
I am not sure it is such a breaking change, clients that don't support that could just ignore the property.

In any cases`filter` should be kept to filtering only, making this argument change the content of the response looks like a very counter intuitive thing to do. Furthermore you could want to have the value in the response while filtering the result (because it could be required later in the client or because the filter could be augmented to filter on other properties). What you propose does not make that possible.

Reply:
I agree with @blacksponge. Adding a new field in the response shouldn't be considered breaking.

But why is it `1` and `0` rather than `true` and `false`?

Another option to consider is to have a "type" field instead, with the two possible values `delegated` and `undelegated`.

Reply:
I changed undelegated to be a boolean.

Reply:
1. If you add a property, is that not a breaking change? Should the client just ignore unknown properties according to RPC?
2. In the document it still says >>`"undelegated"`: `1` if the test is undelegated, `0` otherwise.<<

Reply:
> If you add a property, is that not a breaking change? Should the client just ignore unknown properties according to RPC?

I think the client should ignore unknown properties, yes. This allows to extend the API without breaking.

> In the document it still says >>"undelegated": 1 if the test is undelegated, 0 otherwise.<<

should be fixed.

Reply:
> > If you add a property, is that not a breaking change? Should the client just ignore unknown properties according to RPC?
> 
> I think the client should ignore unknown properties, yes. This allows to extend the API without breaking.

In the API document it says
> * Extra top-level properties in request objects are allowed but ignored.
> * Extra properties in the `"params"` object are allowed for some methods but ignored for others.

Neither in the API document  nor in the [JSON-RPC 2.0] specification I can find any support for unknown properties in responses. If I am wrong, it is fine.

[JSON-RPC 2.0]:  https://www.jsonrpc.org/specification

Reply:
It's very reasonable to assume that clients only look for properties that they know about in responses. If we add a new property to a response without changing the semantics of the properties, the client should is able to continue as if nothing happened. This kind of a change should result in an incremented second digit in semantic versioning.

Reply:
I do not see any problem to have it like that, but then the API document should state that since it is not clear from JSON-RPC 2.0.